### PR TITLE
Cleanup the framework code

### PR DESCRIPTION
**Pull request by simonxiaoss**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,4 +7,7 @@ UploadStatusFile*
 *.azure.env
 *-buildlogs.zip
 .vscode/settings.json
-*.exe
\ No newline at end of file
+*.exe
+Tools/kvp_*
```
Comment (juhlee-microsoft) [inline .gitignore]: Those are required binaries. Any reason you put ignore list?

File: Libraries/Azure.psm1 [hunk 2]
```diff
@@ -583,6 +585,7 @@ Function CreateResourceGroup([string]$RGName, $location) {
         }
         catch {
             $retValue = $false
+            LogErr "Exception in CreateResourceGroup"
```
Comment (juhlee-microsoft) [inline Libraries/Azure.psm1]: There is recommended programming style in catch block. Please add 'Exception reason and line number'. You can find the working code from everywhere.

File: Libraries/Azure.psm1 [hunk 3]
```diff
@@ -637,6 +640,7 @@ Function CreateResourceGroupDeployment([string]$RGName, $location, $setupType, $
         }
         catch {
             $retValue = $false
+            LogErr "Exception in CreateResourceGroupDeployment"
```
Comment (juhlee-microsoft) [inline Libraries/Azure.psm1]: The same comment applied for exceptional reason and line number.

File: .gitignore (hunk 1 above)
Comment (simonxiaoss) [reply .gitignore]: Yes they are required by the test; but the git repo does not host them, so ignore.

File: Libraries/Azure.psm1 (hunk 2 above)
Comment (simonxiaoss) [reply Libraries/Azure.psm1]: fixed; but need future rework to refine the exception handling

File: Libraries/Azure.psm1 (hunk 3 above)
Comment (simonxiaoss) [reply Libraries/Azure.psm1]: fixed; but need future rework to refine the exception handling

**Pull request merged**